service/src/test/java/bio/terra/cbas/dao/TestMethodDao.java

```diff
@@ -50,22 +58,40 @@ class TestMethodDao {
           methodSource,
           workspaceId);
 
+  @Container
+  static JdbcDatabaseContainer postgres =
+      new PostgreSQLContainer("postgres:14")
+          .withDatabaseName("test_db")
+          .withUsername("test_user")
+          .withPassword("test_password");
+
+  @DynamicPropertySource
+  static void postgresProperties(DynamicPropertyRegistry registry) {
+    registry.add("spring.datasource.jdbc-url", postgres::getJdbcUrl);
+    registry.add("spring.datasource.username", postgres::getUsername);
+    registry.add("spring.datasource.password", postgres::getPassword);
+  }
+
   @BeforeAll
-  void setUp() {
+  static void setup() {
+    postgres.start();
```

Comment:
This means that each test reuses the same test container as the previous one... is that deliberate? I guess it's more efficient if the tests never interfere with each other but feels potentially risky?

Reply:
Never mind - the aftereach makes sure they are all independent!